tests/subsys/net/lib/fota_download/src/test_fota_download.c

```diff
@@ -11,24 +12,19 @@
 #include <fw_info.h>
 #include <pm_config.h>
 #include <fota_download.h>
+#include "test_fota_download_common.h"
 
-/* Create buffer which we will fill with strings to test with.
- * This is needed since 'dfu_ctx_Mcuboot_set_b1_file` will modify its
- * 'file' parameter, hence we can not provide it raw c strings as they
- * are stored in read-only memory
+/* fota_download_start may modify the resource locator passed to it, so we cannot directly pass
+ * test constants. Instead, we copy them to a modifiable buffer first.
  */
-char buf[1024];
+static char buf[1024];
 
-#define S0_S1 "s0 s1"
-#define S0 "s0"
-#define S1 "s1"
-#define NO_SPACE "s0s1"
+#include "test_fota_download_common.h"
+#define BASE_DOMAIN "something.com"
 #define NO_TLS -1
-
 #define ARBITRARY_IMAGE_OFFSET 512
 
 /* Stubs and mocks */
-bool dfu_ctx_mcuboot_set_b1_file__s0_active;
 const char *download_client_start_file;
```

Comment:
Should `download_client_start_file` and `dfu_ctx_mcuboot_set_b1_file__update` need to be public?  I know your PR didn't add these, but maybe should be static anyway.